### PR TITLE
Updating the ml_utils file

### PR DESCRIPTION
**Pull request by technoapurva**
PS: Please ignore the minor changes in the ipynb file that is also part of this commit.

Update util files with some changes for Regularization 
- Added batch norm for model building 
- Added ylim to all plot functions for better visualizations 
- Added functions to visualize generalization gap and weights 
- Changed load and save model state function so that it doesn't fail with the update of ml_utils.py 
- All changes are backward compatible

### CODE REVIEW

File: utils/ml_utils.py [hunk 1]
```diff
@@ -417,9 +501,16 @@ def get_model_state(model, model_history, time_callback):
...
+        model_state_serialize[key]=(state.weights,state.history,state.times)
+    pickle.dump(model_state_serialize, 
+                open("pickled_objects/{filename}.pickle".format(filename=filename), "wb" ))
     
     
 def load_model_state(filename):
-    return pickle.load(open("pickled_objects/{filename}.pickle".format(filename=filename), "rb" ))
-
+    model_state_serialize=pickle.load(open("pickled_objects/{filename}.pickle".format(filename=filename), "rb" ))
+    model_state_by_regularizers={}
```
Comment (darylchang123) [inline utils/ml_utils.py]: Maybe rename this to be more generic?

File: utils/ml_utils.py [hunk 2]
```diff
@@ -417,9 +501,16 @@ def get_model_state(model, model_history, time_callback):
...
+    for key,state in model_state_serialize.items():
+        model_state_by_regularizers[key]=ModelState(weights=state[0],history=state[1],times=state[2])
+    return model_state_by_regularizers
```
Comment (darylchang123) [inline utils/ml_utils.py]: Ditto here and here

**Review by darylchang123 — APPROVED** [review]
Looks great, thanks for making these changes! My only major comment is, do you want to put the BatchNorm before the reLU units?

Approving in case you'd rather do that in a separate pull request.

File: utils/ml_utils.py (hunk 1 above)
Comment (technoapurva) [reply utils/ml_utils.py]: Sure!

File: utils/ml_utils.py (hunk 2 above)
Comment (technoapurva) [reply utils/ml_utils.py]: Sure!
